toolchain/check/handle_let_and_var.cpp

```diff
@@ -82,6 +83,17 @@ auto HandleParseNode(Context& context, Parse::VariablePatternId node_id)
     subpattern_id = context.node_stack().PopPattern();
   }
   auto type_id = context.insts().Get(subpattern_id).type_id();
+  type_id = context.AsConcreteType(
+      type_id, node_id,
```

Comment:
nit: how do you feel about moving the `Get().type_id()` call into here so that we only set `type_id` once instead of overwriting it?